lib/blake3_dotnet/src/lib.rs

```diff
@@ -93,3 +93,18 @@ pub extern fn blake3_finalize_xof(
   let slice = unsafe { std::slice::from_raw_parts_mut(ptr_out as *mut u8, size as usize) };
   hasher.finalize_xof().fill(slice);
 }
+
+// Finalize the hash, seek to the offset in the hash stream and put the result into output.
+#[no_mangle]
+pub extern fn blake3_finalize_seek_xof(
```

Comment:
Why adding this function instead if using the existing finalize function by moving the input `ptr_out` from C#?

Reply:
That wouldn't work, unless I'm missing something; the current `Finalize` only sees the slice that you pass it (not the underlying buffer), and always copies from position 0 of the hash's output stream to position 0 of that slice. What `blake3_finalize_seek_xof` allows is to specify the hash's output stream starting position.

Say you want to get 32 bytes from 256Gb in the output stream; you don't want to have to do `var buffer = new byte[256*1024*1024*1024]`, pass that to `Finalize`, and then use `buffer[^32..]`. You want to get to those bytes "directly", using a 32 bytes buffer.

This method is needed whenever you need to generate an "infinite" (well, in this case 2^64-1) amount of bytes from a single seed (eg, you call `Update` only once, at the start), for example when you need a good random number generator. You can then use `Finalize(offset, buffer)` to generate the number of bytes you need for the iteration, and you "advance" the stream on your next call by doing `offset += bytesConsumed` before the next `Finalize(offset, buffer)`.

Reply:
But isn't setting the offset `reader.set_position(offset);` actually equivalent to doing `ptr_out += offset`?  (so starting the position where the slice will be filled)

Reply:
No; look at my 256Gb example above. We won't have a pointer to a 256 gigabytes buffer (make that 256 terabytes if that makes it clearer to you) in any cases.

More importantly, that wouldn't achieve the same thing; `ptr_out` points to the **output** buffer, eg, where the hash's stream data **at its current position** will be copied. What we want to choose, is **from where** within the hash's stream that data will be copied.

Using file streams as an analogy, what you propose is
```
string twoHundred56GigabytesFilePath = ...;
var ptr_out = File.ReadAllBytes(twoHundred56GigabytesFilePath);
Console.WriteLine(Convert.ToHexString(ptr_out[^32..]);
```

With this PR, you can instead do:
```
string twoHundred56GigabytesFilePath = ...;
using var fs = File.OpenRead(twoHundred56GigabytesFilePath);
fs.(256*1024*1024*1024-32, SeekOrigin.Begin);
var ptr_out = new byte[32];
fs.Read(ptr_out, 0, 32); // would need to Read until all 32 bytes are read, but you get the idea
```

Hope this makes it clearer.

Reply:
Take a look at https://github.com/BLAKE3-team/BLAKE3/issues/73#issuecomment-603592191, they talk about something similar.

Reply:
Yep ok, sorry, didn't read correctly your previous post, it is merged. I will update the native binaries after. Thanks!